internal/character/march7th/march7th.go

```diff
@@ -0,0 +1,58 @@
+package march7th
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/target/character"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+func init() {
+	character.Register(key.March7th, character.Config{
+		Create:     NewInstance,
+		Rarity:     4,
+		Element:    model.DamageType_ICE,
+		Path:       model.Path_PRESERVATION,
+		MaxEnergy:  120,
+		Promotions: promotions,
+		Traces:     traces,
+		SkillInfo: character.SkillInfo{
+			Attack: character.Attack{
+				SPAdd:      1,
+				TargetType: model.TargetType_ENEMIES,
+			},
+			Skill: character.Skill{
+				SPNeed:     1,
+				TargetType: model.TargetType_ALLIES,
+			},
+			Ult: character.Ult{
+				TargetType: model.TargetType_ENEMIES,
+			},
+			Technique: character.Technique{
+				TargetType: model.TargetType_ENEMIES,
+				IsAttack:   true,
+			},
+		},
+	})
+}
+
+type char struct {
+	engine engine.Engine
+	id     key.TargetID
+	info   info.Character
+}
+
+func NewInstance(engine engine.Engine, id key.TargetID, charInfo info.Character) info.CharInstance {
+	c := &char{
+		engine: engine,
+		id:     id,
+		info:   charInfo,
+	}
+	c.initTalent()
+
+	engine.Events().BattleStart.Subscribe(c.addE2Shield)
```

Comment:
missing E2 check. Current implementation will always add E2 shield regardless if E2 is enabled or not